### ISSUE TITLE
`Que.execute` converts `true` into `false` when using ActiveRecord 5

### ISSUE THREAD

**Issue by owst** [chanks/que#199]
When using Que with ActiveRecord 5, 
```
p Que.execute('SELECT true AS result')
``` 
gives 
```
[{"result"=>false}]
```
which is a little surprising.

This appears to be due to [`cast_result`](https://github.com/chanks/que/blob/496aa68453a6a08c27a8725ae4e3c34e0b74c3b8/lib/que/adapters/base.rb#L110-L124) changing the value of "result" to `'t' == true` which is `false`. I added `puts "#{value.inspect} -> #{hash[field].inspect}"` after [this line](https://github.com/chanks/que/blob/496aa68453a6a08c27a8725ae4e3c34e0b74c3b8/lib/que/adapters/base.rb#L117) to see what was going on. I assume something has changed in AR such that the conversion of `'t' => true` has already happened, before `cast_result` is called.

If I pin ActiveRecord back to major version 4, I get `[{"result"=>true}]` as expected.

To reproduce, change 5 to 4 in the following:
```
require 'bundler/inline'
gemfile do
  source 'https://rubygems.org'
  gem 'pg'
  gem 'que'
  gem 'activerecord', '~> 5'
end

require 'pg'
require 'active_record'

ActiveRecord::Base.establish_connection(
  adapter: 'postgresql', host: 'localhost',
  user: ENV['DB_USERNAME'], password: ENV['DB_PASSWORD'], database: ENV['DB_NAME']
)

require 'que'
Que.connection = ActiveRecord

p Que.execute('SELECT true AS result')
```

**Comment by chanks** [comment]
Hey! Thanks for the bug report. I just pushed de47b6f, which should resolve this. If that works for you, let me know and I can cut a new release.

This bug is a little weird, since the goal of working directly with PG::Connection objects is to avoid these sort of ORM-specific changes, but I guess AR is doing something different to configure the connection when it's instantiated.

**Comment by owst** [comment]
Hi @chanks - thanks for the quick fix, it works for me!

**Comment by chanks** [comment]
Hey, I haven't forgotten about this, I'm just trying to figure out how to reproduce this outside of Rails so I can spec it better and be sure of what's going on.

**Comment by owst** [comment]
No worries @chanks, thanks for the update

**Comment by chanks** [comment]
Still not sure what Rails is doing to make this fail, but I have a test case in the 1.0 branch so I'm happy. Thanks again @owst!

**Comment by chanks** [comment]
Released version 0.14.1.
